cmd/token.go

```diff
@@ -370,3 +385,53 @@ func createToken(cmd *cobra.Command, args []string) error {
 	fmt.Println(tok)
 	return nil
 }
+
+// Fetch a token for a given Pelican URL using the TokenGenerator
+func fetchToken(cmd *cobra.Command, args []string) error {
+	err := config.InitClient()
+	if err != nil {
+		return errors.Wrapf(err, "unable to initialize client config")
+	}
+
+	rawUrl := args[0]
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	pUrl, pUrlErr := client.ParseRemoteAsPUrl(ctx, rawUrl)
+	if pUrlErr != nil {
+		return errors.Wrapf(pUrlErr, "failed to parse Pelican URL")
+	}
+
+	// Determine whether to fetch a write token
+	write, _ := cmd.Flags().GetBool("write")
+	modify, _ := cmd.Flags().GetBool("modify")
+
+	var oper config.TokenOperation
+	oper = config.TokenRead
+	method := http.MethodGet
+
+	if write {
+		oper = config.TokenWrite
+		method = http.MethodPut
+	}
+	if modify {
+		oper = config.TokenDelete
+		method = http.MethodPut
+	}
```

Comment:
This assumes token operations are mutually exclusive and that you'll never want a token for more than one operation. If that's the case, the behavior should be documented in in the CLI and the flags should be marked as mutually exclusive.